docs/csharp/fundamentals/tutorials/file-based-programs.md

```diff
@@ -0,0 +1,246 @@
+
+:::code language="csharp" source="./snippets/file-based-programs/AsciiArt.cs" id="RootCommand":::
+
+Next, add the code to parse the command line arguments and handle any errors. This code validates the command line arguments and stores parsed arguments in the <xref:System.CommandLine.ParseResult?displayProperty=nameWithType> object:
+
+:::code language="csharp" source="./snippets/file-based-programs/AsciiArt.cs" id="ParseAndValidate":::
+
+The preceding code validates all command line arguments. If the validation fails, errors are written to the console, and the app exits.
+
+## Use parsed command line results
+
+Now, finish the app to use the parsed options and write the output. First, define a record to hold the parsed options. File-based apps can include type declarations, like records. They must be after all top-level statements and local functions. Add a `record` declaration to store the messages and the delay option value:
+
+:::code language="csharp" source="./snippets/file-based-programs/AsciiArt.cs" id="Record":::
+
+Now that you declared the record to store those results, add a local function to process the parse results and store the values in an instance of the record. Add the following local function before the record declaration. This method handles both command line arguments and standard input, and returns a new record instance:
+
+:::code language="csharp" source="./snippets/file-based-programs/AsciiArt.cs" id="ProcessParsedArgs":::
+
+Next, create a local function to write the ASCII art with the specified delay. This function writes each message in the record with the specified delay between each message:
+
+:::code language="csharp" source="./snippets/file-based-programs/AsciiArt.cs" id="WriteAscii":::
+
+Finally, replace the `if` clause you wrote earlier with the following code to process the command line arguments and write the output:
+
+:::code language="csharp" source="./snippets/file-based-programs/AsciiArt.cs" id="InvokeCommand":::
+
+## Test the final application
+
+Test the application by running several different commands. If you have trouble, here's the finished sample to compare with what you built:
+
+:::code language="csharp" source="./snippets/file-based-programs/AsciiArt.cs":::
+
+In this tutorial, you learned to build a file-based program, where you build the program in a single C# file. These programs don't use a project file, and can use the `#!` directive on unix systems. Learners can create these programs after trying our [online tutorials](../../tour-of-csharp/tutorials/hello-world.md) and before building larger project-based programs. File based programs are also a great platform for command line utilities.
+
```

Comment:
Considering the code is only around 65 lines of code (with the //snippet comments removed) you could add a section to the end with all the code as an example. If you want to do that you need to prep the file first. You can't just reference the entire file because it will contain the //snippet codes. Add a `// <all>` and `// </all>` snippet at the top and bottom of the file, respectively. Then reference the `all` snippet in the ::: code reference. This erases all of the other //snippet comments in the output.

Reply:
In the end, I copied the source file and trimmed the tags from one of them.